pontoon/sync/vcs/repositories.py

```diff
@@ -40,8 +42,16 @@ def pull(self, source=None, target=None):
         code, output, error = execute(command, target)
 
         if code == 0:
+
             log.debug("Git: Repository at " + source + " updated.")
 
+            if branch:
+                command = ["git", "checkout", branch]
+                code, output, error = execute(command)
+
+                if code == 0:
+                    log.debug("Git: Branch " + branch + " checked out.")
```

Comment:
We should raise `PullFromRepositoryException` if code != 0, otherwise Pontoon will silently ignore the branch and fallback to the default.